src/components/organisms/AssetActions/Compute/index.tsx

```diff
@@ -0,0 +1,300 @@
+    trustedAlgorithmList: publisherTrustedAlgorithm[]
+  ): SearchQuery {
+    let algoQuerry = ''
+    trustedAlgorithmList.forEach((trusteAlgo) => {
+      algoQuerry += `id:"${trusteAlgo.did}" OR `
+    })
+    if (trustedAlgorithmList.length > 1) {
+      algoQuerry = algoQuerry.substring(0, algoQuerry.length - 3)
+    }
+    const algorithmQuery =
+      trustedAlgorithmList.length > 0 ? `(${algoQuerry}) AND` : ``
+    const query = {
+      page: 1,
+      query: {
+        query_string: {
+          query: `${algorithmQuery} service.attributes.main.type:algorithm AND price.type:exchange -isInPurgatory:true`
+        }
+      },
+      sort: { created: -1 }
+    }
+    return query
+  }
+
+  async function getAlgorithmList(): Promise<AssetSelectionAsset[]> {
+    const source = axios.CancelToken.source()
+    const computeService = ddo.findServiceByType('compute')
+    let algorithmSelectionList: AssetSelectionAsset[]
+    if (
+      computeService.attributes.main.privacy.publisherTrustedAlgorithms
+        .length === 0 &&
+      !computeService.attributes.main.privacy.allowAllPublishedAlgorithms
+    ) {
+      algorithmSelectionList = []
+    } else {
+      const gueryResults = await queryMetadata(
+        getQuerryString(
+          computeService.attributes.main.privacy.publisherTrustedAlgorithms
+        ),
+        config.metadataCacheUri,
+        source.token
+      )
+      setDdoAlgorithmList(gueryResults.results)
+      algorithmSelectionList = await transformDDOToAssetSelection(
+        gueryResults.results,
+        config.metadataCacheUri,
+        []
+      )
+    }
+    return algorithmSelectionList
+  }
+
+  useEffect(() => {
+    if (!ddo) return
+    getAlgorithmList().then((algorithms) => {
+      setAlgorithmList(algorithms)
+    })
+  }, [ddo])
+
+  useEffect(() => {
+    if (!ocean || !accountId) return
+    checkPreviousOrders(ddo, 'compute')
+  }, [ocean, ddo, accountId])
+
+  useEffect(() => {
+    if (!ocean || !accountId || !selectedAlgorithmAsset) return
+    checkPreviousOrders(selectedAlgorithmAsset, 'access')
+    checkAssetDTBalance(selectedAlgorithmAsset)
+  }, [selectedAlgorithmAsset, ocean, accountId])
+
+  const startJob = async (algorithmId: string) => {
+    try {
+      if (!ocean) return
+
+      setIsJobStarting(true)
+      setIsPublished(false)
+      setError('')
+
+      const computeService = ddo.findServiceByType('compute')
+      const serviceAlgo = selectedAlgorithmAsset.findServiceByType('access')
+
+      !hasPreviousDatasetOrder &&
+        !hasDatatoken &&
+        (await buyDT('1', (ddo as DDO).price, ddo))
+
+      !hasPreviousAlgorithmOrder &&
+        !hasAlgoAssetDatatoken &&
+        (await buyDT(
+          '1',
+          (selectedAlgorithmAsset as DDO).price,
+          selectedAlgorithmAsset
+        ))
+
+      const allowed = await ocean.compute.isOrderable(
+        ddo.id,
+        computeService.index
+      )
+
+      console.log('is dataset orderable ?', allowed)
```

Comment:
would be super useful to add some debug logging, so after every step we could output like:

```js
Logger.log('[compute] isOrderable? ', isOrderable)
```